frontend/src/lib/Login.svelte

```diff
@@ -0,0 +1,54 @@
+<script lang="ts">
+  import { authToken } from './stores';
+
+  let username = '';
+  let password = '';
+  let error: string | null = null;
+
+  async function login() {
+    error = null;
+    try {
+      const response = await fetch('/api/login', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ username, password }),
+      });
+
+      const data = await response.json();
+      if (!response.ok) {
+        throw new Error(data.message || 'Login failed');
+      }
+      authToken.set(data.token);
```

Comment:
data.result.token, not data.token.


Add also the handling of the refresh_token.